live-flowchart/logger.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * @global Logger
+ * Logging helper
+ */
+var Logger = {
+
+  /*
+   * @readonly logDomObj
+   * The HTML DOM Object to log onto
+   */
+  logDomObj: document.querySelector('#log'),
+
+  /*
+   * @method debug
+   * Debug log
+   * @param {String} message The message to be logged
+   */
+  debug: function debug(message) {
+    this.writeLog(message, { debug: true });
+  },
+
+  /*
+   * @method info
+   * Info log
+   * @param {String} message The message to be logged
+   */
+  info: function info(message) {
+    this.writeLog(message, { info: true });
+  },
+
+  /*
+   * @method log
+   * Default log
+   * @param {String} message The message to be logged
+   */
+  log: function log(message) {
+    this.writeLog(message);
+  },
+
+  /*
+   * @method warn
+   * Warning log
+   * @param {String} message The message to be logged
+   */
+  warn: function warn(message) {
+    this.writeLog(message, { warn: true });
+  },
+
+  /*
+   * @method error
+   * Error log
+   * @param {String} message The message to be logged
+   */
+  error: function error(message) {
+    this.writeLog(message, { error: true });
+  },
+
+  /*
+   * @method log
+   * Log onto the browser console and the HTML log
+   * @param {String} message The message to be logged
+   * @param {Object} level The log level (error, warn, info, log, debug) Default: level.log
+   */
+  writeLog: function writeLog(message, level) {
+    var logMessage;
+    var newLine;
+
+    logMessage = document.createElement('div');
+
+    if (message) {
+      // add message to the HTML node
+      logMessage.innerHTML = message;
+
+      // new like check
+      if (Object.prototype.toString.call(message) === '[object String]'
```

Comment:
Yea, `typeof message === "string"` is way more readable, and do the same job in this context. Thanks.